### DISCUSSION TITLE
Record 1 minute every 30 minutes, only between the hours of 05:30 and 14:00

### DISCUSSION THREAD

**Discussion by andrewSharo** [gopro/labs#944] [General]
Hi All,

As described in the title, I would like to record 1 minute every 30 minutes, only between the hours of 05:30 and 14:00. I didn't see a prior discussion on this exact scenario, so thought I would share my current plan. I just started today with GoPro labs and this is pretty fun!

My QR: >14:00!05:30NmV0r1080p24fW0e0hS0dR0aSoB0!S!58E!1740!R

Pseudocode:

If (current time >= 14:00)
{
	Sleep until 5:30
}
Adjust video settings
Record for 58 seconds (this gives 1 minute of recording for some odd reason)
Sleep for 1740 seconds (30 minutes minus the 1 minute that was recorded)
Repeat

I'm going to run it right now and see if this works over the next day. 

If you have suggestions for improving this code, I'm all ears!

Also I would like to note that including the leading 0 in times appears to be critical (i.e., 05:30 NOT 5:30).

Andrew

**Comment by dnewman-gpsw** [comment]
A few of improvements.
1) Don't do settings for every capture, while this will work, it is increasing the failure risk.  All of this mV0r1080p24fW0e0hS0dR0aSoB0 should be in its own command. Also add dV -  default Video.  So use ```mV0dVr1080p24fW0e0hS0dR0aSoB0```
2) Typo !1740 is not a command. New simplified script  ```>14:00!05:30N!S!58E!1N!1739R```
Note: I added !1N after !58E, as the current H11 and H12 firmware has a rare bug which causes the camera to sometimes not to sleep, and the each pause fixes that.  It will be fixed in a future release.

The 58 seconds issue, is the start (!S) command has a 2 second delay built in.  In more recent releases you can control the start delay in milliseconds. e.g. ```*STDL=500```  for 0.5s.  The start delay was added to prevent command being missed as capture starts are a heavy camera load, but 2s is overkill.

**Reply by andrewSharo** [reply]
Hi David,

Thanks so much for these suggestions! I will incorporate them. I should have noted, I'm using a Hero10 Black. One question: When you say that I should have the settings as their own command, should I create two qr codes? I would then show the GoPro first the settings qr code, and then the simplified script qr code? Let me know if that's correct.

I also realized that if I were to start this script between midnight and 5:30, it would probably start to record right away. That's not an issue for me, but wanted to note that in case others wanted to re-use this code. I imagine this could be fixed by instead writing: `>14:00!05:30N<5:30!5:30N!S!58E!1N!1739R`

Andrew

**Reply by dnewman-gpsw** [reply]
Yes, two QR Code is best if you want more robust operations.
